pkg/runtime/wfengine/workflow.go

```diff
@@ -375,21 +390,34 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	// will trigger this callback channel.
 	callback := make(chan bool)
 	wi.Properties[CallbackChannelProperty] = callback
+	// Request to execute workflow
+	start := time.Now()
 	err = wf.scheduler.ScheduleWorkflow(ctx, wi)
 	if err != nil {
 		if errors.Is(err, context.DeadlineExceeded) {
+			// Workflow execution scheduling request failed with recoverable error, record metrics.
+			diag.DefaultWorkflowMonitoring.ExecutionCount(ctx, diag.Workflow, diag.StatusRetryable)
 			return newRecoverableError(fmt.Errorf("timed-out trying to schedule a workflow execution - this can happen if there are too many in-flight workflows or if the workflow engine isn't running: %w", err))
 		}
+		// Workflow execution scheduling request failed with recoverable error, record metrics.
+		diag.DefaultWorkflowMonitoring.ExecutionCount(ctx, diag.Workflow, diag.StatusRetryable)
```

Comment:
If the scheduler has failed, then that would mean that workflow execution has not even started. It couldn't be scheduled only. So probably no need to record metrics in case scheduler returns error